api/src/main/java/org/openmrs/layout/name/NameSupport.java

```diff
@@ -9,16 +9,28 @@
  */
 package org.openmrs.layout.name;
 
-import org.openmrs.api.APIException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.GlobalProperty;
+import org.openmrs.api.GlobalPropertyListener;
 import org.openmrs.api.context.Context;
 import org.openmrs.layout.LayoutSupport;
+import org.openmrs.layout.address.AddressSupport;
+import org.openmrs.serialization.SerializationException;
+import org.openmrs.util.OpenmrsConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * @since 1.12
  */
-public class NameSupport extends LayoutSupport<NameTemplate> {
+public class NameSupport extends LayoutSupport<NameTemplate> implements GlobalPropertyListener {
 	
 	private static NameSupport singleton;
+	private boolean initialized = false;
+	private static final Logger log = LoggerFactory.getLogger(AddressSupport.class);
```

Comment:
Static stuff should be initialized before instance stuff. Also, the logger here is pointing at the wrong class.